tests/Fixer/PhpUnit/PhpUnitDataProviderNameFixerTest.php

```diff
@@ -396,5 +396,27 @@ public function dtPrvdr() {}
                 $config,
             ];
         }
+
+        yield 'abstract data provider' => [
+            '<?php
+abstract class FooTest extends TestCase {
+    /**
+     * @dataProvider provideCustomCases
+     */
+    public function testFoo() {}
+    abstract public function provideCustomCases();
+}',
+        ];
+
+        yield 'abstract test' => [
+            '<?php
+abstract class FooTest extends TestCase {
+    /**
+     * @dataProvider provideCustomCases
+     */
+    abstract public function testFoo();
+    public function provideCustomCases() {}
```

Comment:
I _do_ see your point here.
Yet, even if the method is not abstract, it can be overridden.
As fixer is already marked as risky (and extension of riskiness description is very good!), at short glance I would rather expect it to fix as much cases as possible, rather than limiting some fixes.

Imagine the case when someone creates the `BazTest extends FooTest` (in other file). Even in your case,  and that BazTest also having `testFoo` and (no longer abstract) `provideCustomCases`. Will `provideCustomCases` be renamed?
If yest for BazTest but no for FooTest, then I do not see value of this PR

Reply:
I don't see how we add in this version, only as a configuration option, of course, if we agree that it is worth having (I do not understand the value of this change).